py4DSTEM/process/diffraction/crystal.py

```diff
@@ -779,17 +807,27 @@ def generate_diffraction_pattern(
 
         # Calculate excitation errors
         if foil_normal is None:
-            sg = self.excitation_errors(g)
+            sg = self.excitation_errors(
+                g,
+                precession_angle_degrees=precession_angle_degrees,
+            )
```

Comment:
User should be able to specify number of precession steps

Reply:
I'm planning to replace this with an analytic function, so my preference is to leave this as is for now.